### PR TITLE
ARROW-2517: [Java] Add list<decimal> writer

### PR DESCRIPTION
**Pull request by pudidic**
Apache Arrow have writer interface for list of decimal without implementation. It will follow the current interface and will infer the scale from BigDecimal or DecimalHolder if it can.

cc @wesm

### CODE REVIEW

**Review by xhochy — CHANGES_REQUESTED** [review]
The unit tests fail with 

```
[ERROR] Failures: 
[ERROR]   TestComplexWriter.listDecimalType:285 expected:<1> but was:<972208782254614709842509301809153>
```

**Comment by pudidic** [comment]
Fixed wrong test code. Thanks for feedback.

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
cc @jacques-n @icexelloss @siddharthteotia @BryanCutler

File: java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java [hunk 1]
```diff
@@ -45,6 +46,7 @@
    * @return the corresponding field writer
    */
   abstract protected FieldWriter getWriter(MinorType type);
+  abstract protected FieldWriter getWriter(MinorType type, ArrowType arrowType);
```
Comment (BryanCutler) [inline java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java]: Could you elaborate on why you need to add the `arrowType`?

**Review by BryanCutler — COMMENTED** [review]
Thanks for the PR @pudidic.  The linked JIRA is wrong though, I think it should be ARROW-2517.  I don't normally use the writer interface, so I only made a quick pass but I'm not too sure why you need to use the ArrowType when setting the writer.  I believe there is already an interface that allows you to set the scale and precision.  Could you elaborate on that a little?  Maybe @siddharthteotia can take a closer look.

**Review by siddharthteotia — APPROVED** [review]
LGTM

**Pull request merged**